cmd/clusterctl/client/repository/components_test.go

```diff
@@ -182,12 +202,136 @@ func Test_fixTargetNamespace(t *testing.T) {
+						"metadata": map[string]interface{}{
+							"annotations": map[string]interface{}{
+								"cert-manager.io/inject-ca-from": "bar/capm3-serving-cert",
+							},
+							"creationTimestamp": nil,
+							"name":              "capm3-mutating-webhook-configuration",
+						},
+						"webhooks": []interface{}{
+							map[string]interface{}{
+								"name":                    "",
+								"admissionReviewVersions": nil,
+								"clientConfig": map[string]interface{}{
+									"service": map[string]interface{}{
+										"name":      "capm3-webhook-service",
+										"path":      "/mutate-infrastructure-cluster-x-k8s-io-v1alpha4-metal3cluster",
+										"namespace": "bar",
+									},
+									"caBundle": "Cg==",
+								},
+								"sideEffects": nil,
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "fix crd webhook namespace",
+			objs: []unstructured.Unstructured{
+				{
+					Object: map[string]interface{}{
+						"apiVersion": "apiextensions.k8s.io/v1",
+						"kind":       "CustomResourceDefinition",
+						"spec": map[string]interface{}{
+							"conversion": map[string]interface{}{
+								"strategy": "Webhook",
+								"webhook": map[string]interface{}{
+									"clientConfig": map[string]interface{}{
+										"caBundle": "Cg==",
+										"service": map[string]interface{}{
+											"name":      "capa-webhook-service",
+											"namespace": "capi-webhook-system",
+											"path":      "/convert",
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			targetNamespace: "bar",
+			want: []unstructured.Unstructured{
+				{
+					Object: map[string]interface{}{
+						"apiVersion": "apiextensions.k8s.io/v1",
+						"kind":       "CustomResourceDefinition",
+						"metadata": map[string]interface{}{
+							"creationTimestamp": nil,
+						},
+						"spec": map[string]interface{}{
+							"group":    "",
+							"names":    map[string]interface{}{"plural": "", "kind": ""},
+							"scope":    "",
+							"versions": nil,
+							"conversion": map[string]interface{}{
+								"strategy": "Webhook",
+								"webhook": map[string]interface{}{
+									"conversionReviewVersions": nil,
+									"clientConfig": map[string]interface{}{
+										"caBundle": "Cg==",
+										"service": map[string]interface{}{
+											"name":      "capa-webhook-service",
+											"namespace": "bar",
+											"path":      "/convert",
+										},
+									},
+								},
+							},
+						},
+						"status": map[string]interface{}{
+							"storedVersions": nil,
+							"conditions":     nil,
+							"acceptedNames":  map[string]interface{}{"kind": "", "plural": ""},
+						},
+					},
+				},
+			},
+		},
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			g := NewWithT(t)
 
-			got := fixTargetNamespace(tt.args.objs, tt.args.targetNamespace)
+			got, err := fixTargetNamespace(tt.objs, tt.targetNamespace)
+			g.Expect(err).To(Succeed())
```

Comment:
```suggestion
			g.Expect(err).ToNot(HaveOccurred())
```
nit: We usually use that one when we just validate an `err`. We use `Succeed` when there's a function call in `Expect()`